src/Format.cpp

```diff
@@ -11,42 +11,35 @@
 
 ByteString format::URLEncode(ByteString source)
 {
-	char * src = (char *)source.c_str();
-	char * dst = new char[(source.length()*3)+2];
-	std::fill(dst, dst+(source.length()*3)+2, 0);
+	ByteString finalString;
 
-	char *d;
-	for (d = dst; *d; d++) ;
-
-	for (unsigned char *s = (unsigned char *)src; *s; s++)
+	for (auto& s : source)
 	{
-		if ((*s>='0' && *s<='9') ||
-		        (*s>='a' && *s<='z') ||
-		        (*s>='A' && *s<='Z'))
-			*(d++) = *s;
+		if ((s >= '0' && s <= '9') ||
+			(s >= 'a' && s <= 'z') ||
+			(s >= 'A' && s <= 'Z'))
+			finalString.insert(finalString.end(), s);
 		else
 		{
-			*(d++) = '%';
-			*(d++) = hex[*s>>4];
-			*(d++) = hex[*s&15];
+			finalString.insert(finalString.end(), '%');
+			finalString.insert(finalString.end(), s >> 4);
+			finalString.insert(finalString.end(), s & 15);
 		}
 	}
-	*d = 0;
 
-	ByteString finalString(dst);
-	delete[] dst;
 	return finalString;
 }
 
 ByteString format::UnixtimeToDate(time_t unixtime, ByteString dateFormat)
 {
 	struct tm * timeData;
-	char buffer[128];
+	ByteString buffer;
+	buffer.reserve(128);
 
 	timeData = localtime(&unixtime);
 
-	strftime(buffer, 128, dateFormat.c_str(), timeData);
-	return ByteString(buffer);
+	strftime(&buffer[0], 128, dateFormat.c_str(), timeData);
+	return buffer;
```

Comment:
This is broken. The resulting string doesn't have a `size` of 128 but merely the `capacity`...